src/module_PI/Raspberry_PI/main/EmailNotifier.java

```diff
@@ -0,0 +1,239 @@
+package module_PI.Raspberry_PI.main;
+
+import java.io.*;
+import java.util.*;
+
+import javax.activation.*;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+public class EmailNotifier {
+
+	private String gmailUsername;
+	private String gmailPassword;
+
+	private String recipientEmailAddress;
+	private String emailSubject = "Calico Home Security";
+	private String emailMessage;
+	private String attachmentFilePath;
+
+	private Properties mailSessionProperties;
+	private Session javaMailSession;
+	private Message javaMailMessage;
+	private BodyPart messageBodyPart;
+	private Multipart multipart;
+	private DataSource attachmentDataSource;
+
+	public EmailNotifier(String recipientEmailAddress, String message) {
+		this(recipientEmailAddress, message, null);
+	}
+
+	public EmailNotifier(String recipientEmailAddress, String message, String pathToAttachment) {
+		this.loadGmailCredentials();
+		this.setRecipientEmailAddress(recipientEmailAddress);
+		this.emailMessage = message;
+		this.setAttachmentFilePath(pathToAttachment);
+		buildPropertiesAndCreateSession();
+	}
+
+	private void loadGmailCredentials() {
+		try {
+			Properties gmailProperties = new Properties();
+			FileInputStream in = new FileInputStream("gmailCredentials");
```

Comment:
Ohhh I see what you're saying. I'll take another look at this tonight when I have time. I was wondering about the visibility of that method because it seems like just another object with standard visibility.